lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -23,11 +23,12 @@ def list(rgroup = configuration.resource_group)
         JSON.parse(response)['value'].map { |hash| model_class.new(hash) }
       end
 
-      def list_all
+      def list_all(filter = {})
         url = build_url
         url = yield(url) || url if block_given?
         response = rest_get(url)
-        JSON.parse(response)['value'].map { |hash| model_class.new(hash) }
+        results = JSON.parse(response)['value'].map { |hash| model_class.new(hash) }
+        results.select { |obj| filter.all? { |k, v| obj.public_send(k) == v } }
```

Comment:
Check `filter` whether empty (better default to `nil` instead of `{}`) before `select` for performance sake.